script.js

```diff
@@ -0,0 +1,40 @@
+$(document).ready(function(){
+	counter = 0;
+	myInterval = 0;
+
+	$('#submit').on('click', function(e){
+ 		e.preventDefault();
+ 		var input = $('#input').val();
+				$('#input').val('');
+	$('.container').hide(1000);
+
+ 	$.get('http://www.reddit.com/search.json', {
+ 		q: input
+ 		}).done(function(data){
+
+ 			var searchResult = data.data.children.map(function(item){
+ 				if (item.data.preview !== undefined){
```

Comment:
You're trying to filter here, not map. Map takes every item in an array and turns it into something else. Filter decides whether or not to include an item in an array. The conditionals mean that you'll be returning undefined sometimes.